cli/flox-rust-sdk/src/providers/build.rs

```diff
@@ -67,13 +73,32 @@ pub enum ManifestBuilderError {
     },
 }
 
+#[derive(Debug)]
 pub enum Output {
     /// A line of stdout output from the build process.
     Stdout(String),
     /// A line of stderr output from the build process.
     Stderr(String),
-    /// The build process has exited with the given status.
-    Exit(ExitStatus),
+    /// The build process has successfully and produced the given [BuildResults].
+    Success(BuildResults),
+    /// The build process has failed with the given exit status.
+    /// On error `flox-build.mk` will not produce a build result file,
+    /// so we don't have build results to return.
+    /// Todo: we may want to recombine [Output::Failure] and [Output::Success]
+    /// eventually if `flox-build.mk` is updated to always produce a build result file,
+    /// e.g. containing error context for the failing builds.
+    Failure(ExitStatus),
+}
+
+#[derive(Debug, PartialEq, Deserialize, Default, derive_more::Deref)]
+pub struct BuildResults(Vec<BuildResult>);
+
+#[derive(Debug, PartialEq, Deserialize)]
+pub struct BuildResult {
+    pub pname: String,
+    pub outputs: HashMap<String, BuiltStorePath>,
+    pub version: String,
+    pub log: BuiltStorePath,
 }
```

Comment:
I'm a bit confused about how this will be used by `publish` to complete #2601 and #2602 when `flox build` and `flox publish` are separate processes, so it doesn't have the `BuildResult` or the name of the persisted temporary file that it was deserialized from. Can you explain the next steps?

Reply:
#2602 is meant to use the build provider to build the published packages in a fresh directory.
The build results of _that_ build are what are being published.